### PR TITLE
Intercom integration

### PR DESCRIPTION
**Pull request by Shpigford**
Includes if/else statements for various ways to reach out. Also, github/discord icons updated to SVG.

### CODE REVIEW

File: app/views/layouts/_sidebar.html.erb [hunk 1]
```diff
@@ -57,9 +57,16 @@
           <%= lucide_icon("megaphone", class: "w-5 h-5 text-gray-500 shrink-0") %>
           <span class="text-gray-900 text-sm">Feedback</span>
         <% end %>
-        <%= link_to "https://link.maybe.co/discord", class: "flex gap-2 items-center hover:bg-gray-50 rounded-lg px-3 py-2" do %>
-          <%= lucide_icon("message-square-more", class: "w-5 h-5 text-gray-500 shrink-0") %>
-          <span class="text-gray-900 text-sm">Contact</span>
+        <% if ENV["SELF_HOSTED"] == "true" %>
```
Comment (zachgoll) [inline app/views/layouts/_sidebar.html.erb:60]:
```suggestion
        <% if self_hosted? %>
```

In all view-related code, I've consolidated this to `self_hosted?` helper in `SelfHostable` just to consolidate changes to one centralized spot 

https://github.com/maybe-finance/maybe/blob/main/app/controllers/concerns/self_hostable.rb

File: app/views/pages/feedback.html.erb [hunk 2]
```diff
@@ -9,17 +9,23 @@
...
       <%= link_to "https://github.com/maybe-finance/maybe/discussions/categories/feature-requests", target: "_blank", rel: "noopener noreferrer", class: "w-1/3 flex flex-col items-center p-4 border border-alpha-black-25 rounded-xl hover:bg-gray-50" do %>
-        <%= image_tag "github-icon.png", class: "w-8 h-8 mb-2" %>
+        <%= image_tag "github-icon.svg", class: "w-8 h-8 mb-2" %>
         <span class="text-sm font-medium text-gray-900">Write a feature request</span>
       <% end %>
-
-      <%= link_to "https://github.com/maybe-finance/maybe/issues/new?assignees=&labels=bug&template=bug_report.md&title=", target: "_blank", rel: "noopener noreferrer", class: "w-1/3 flex flex-col items-center p-4 border border-alpha-black-25 rounded-xl hover:bg-gray-50" do %>
-        <%= image_tag "github-icon.png", class: "w-8 h-8 mb-2" %>
-        <span class="text-sm font-medium text-gray-900">File a bug report</span>
+      <% if ENV["SELF_HOSTED"] == "true" %>
```
Comment (zachgoll) [inline app/views/pages/feedback.html.erb:15]:
```suggestion
      <% if self_hosted? %>
```

File: config/initializers/intercom.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    config.user.custom_data = {
+      family_id: Proc.new { |current_user| current_user.family.id }
+    }
+
+    # == Current company method/variable
+    # The method/variable that contains the current company for the current user,
+    # in your controllers. 'Companies' are generic groupings of users, so this
+    # could be a company, app or group.
+    #
+    # config.company.current = Proc.new { current_company }
```
Comment (zachgoll) [inline config/initializers/intercom.rb:64]: Not entirely sure how this shows up in Intercom UI, but could potentially be repurposed?

```rb
config.company.current = Proc.new { Current.family }
```

Comment (Shpigford) [reply config/initializers/intercom.rb:64]: Oooo, interesting idea. Let me see how that filters down through Intercom.

Comment (Shpigford) [reply config/initializers/intercom.rb]: Done. Super easy.

**Review by zachgoll — APPROVED** [review]

**Pull request merged**